pkg/sidecar/sidecar_test.go

```diff
@@ -252,6 +252,61 @@ func TestGetCmdArgs(t *testing.T) {
 			in:          `-flag1 "value 1`,
 			expectedErr: `missing " in quoted-field`,
 		},
+		// Single quotes are not special to the parser
+		{
+			name:         "Single quotes in double quotes",
+			in:           `-c "echo 'hello world'"`,
+			expectedArgs: []string{"-c", "echo 'hello world'"},
+		},
+		{
+			name:         "Unpaired single quote",
+			in:           `echo Mc'Gougall`,
+			expectedArgs: []string{"echo", "Mc'Gougall"},
+		},
+		// Unlike a shell, spiffe-helper will parse this argument
+		// string without considering the single quoted range as a
+		// single argument.
+		{
+			name:         "single quotes do not protect spaces",
+			in:           `-c 'echo hello world'`,
+			expectedArgs: []string{"-c", "'echo", "hello", "world'"},
+		},
```

Comment:
Test case to illustrate https://github.com/spiffe/spiffe-helper/pull/243#pullrequestreview-2591877988